src/CyBy/Draw/Internal/Graph.idr

```diff
@@ -604,6 +604,34 @@ newEdges t ps = do
   (a1,a2) <- ps
   (\(x,l) => (a1,x,l)) <$> neighboursAsPairs t a2
 
+createStableBond :
+     {k,m : _}
+  -> List (Fin k, Fin m)
+  -> Edge m CDBond
+  -> Maybe $ Edge k CDBond
+createStableBond nsToMerge bnd =
+  let filter := foldMap {m=List (Fin k, Fin m)}
+                  (\(nm,nt) => if nt == bnd.node1 || nt == bnd.node2
+                                 then [(nm, nt)]
+                                 else [])
+                  nsToMerge
+   in case filter of
+        [(nm',_),(nm'',_)] => mkEdge nm' nm'' bnd.label
+        _                  => Nothing
```

Comment:
Some notes:

  * please read and adhere to the style guide before asking for a review. Multi-line lambdas in parentheses should be avoided
  * Be careful with accumulating a list by left-folding over a list: This is always O(n^2). Use either a `SnocList` or the list monad. Both are O(n).
  * In this case, don't use a fold at all. Use `Data.List.filter`.